front/logger/withlogging.ts

```diff
@@ -161,6 +167,34 @@ export function withLogging<T>(
   };
 }
 
+export function dustErrorToApiError(
+  // TODO: add other error codes as we use this method in other places
+  r: Err<Omit<DustError, "code"> & { code: "unauthorized" | "internal_error" }>,
```

Comment:
@Fraggle we generally have these in the API endpoints because each endpoint will react differently to different error codes. No way we'll have one generic big helper for all endpoints and all DustError codes.

In particular the endpoint is in charge of delivering a user friendly message which is rarely the error message per se.

Can you refactor to move this to the endpoints :pray: